DISTRIBUTING.md

```diff
@@ -169,3 +169,352 @@ binary packages. It also includes miscellaneous tools to do things such as
+
+```
+gpg -u julia --armor --detach-sig julia-x.y.z-linux-<arch>.tar.gz
+```
+
+This will produce a corresponding .asc file for each tarball.
+And that's it!
+
+### macOS
+
+Code signing should happen automatically on the macOS buildbots.
+However, it's important to verify that it was successful.
+On a system or virtual machine running macOS, download the .dmg file that was built on
+the buildbots.
+For the sake of example, say that the .dmg file is called `julia-x.y.z-osx.dmg`.
+Run
+
+```
+mkdir ./jlmnt
+hdiutil mount -readonly -mountpoint ./jlmnt julia-x.y.z-osx.dmg
+codesign -v jlmnt/Julia-x.y.app
+```
+
+Be sure to note the name of the mounted disk listed when mounting!
+For the sake of example, we'll assume this is `disk3`.
+If the code signing verification exited successfully, there will be no output from the
+`codesign` step.
+If it was indeed successful, you can detach the .dmg now:
+
+```
+hdiutil detach disk3
+rm -rf ./jlmnt
+```
+
+If you get a message like
+
+> Julia-x.y.app: code object is not signed at all
+
+then you'll need to sign manually.
+
+To sign manually, first retrieve the OS X certificates from the CodeSigning folder
+in the `juliasecure` bucket on AWS.
+Add the .p12 file to your keychain using Keychain.app.
+Ask Elliot Saba (staticfloat) or Alex Arslan (ararslan) for the password for the key.
+Now run
+
+```
+hdiutil convert julia-x.y.z-osx.dmg -format UDRW -o julia-x.y.z-osx_writable.dmg
+mkdir ./jlmnt
+hdiutil mount -mountpoint julia-x.y.z-osx_writable.dmg
+codesign -s "AFB379C0B4CBD9DB9A762797FC2AB5460A2B0DBE" --deep jlmnt/Julia-x.y.app
+```
+
+This may fail with a message like
+
+> Julia-x.y.app: resource fork, Finder information, or similar detritus not allowed
+
+If that's the case, you'll need to remove extraneous attributes:
+
+```
+xattr -cr jlmnt/Julia-x.y.app
+```
+
+Then retry code signing.
+If that produces no errors, retry verification.
+If all is now well, unmount the writable .dmg and convert it back to read-only:
+
+```
+hdiutil detach disk3
+rm -rf ./jlmnt
+hdiutil convert julia-x.y.z-osx_writable.dmg -format UDZO -o julia-x.y.z-osx_fixed.dmg
+```
+
+Verify that the resulting .dmg is in fact fixed by double clicking it.
+If everything looks good, eject it then drop the `_fixed` suffix from the name.
+And that's it!
+
+### Windows
+
+Signing must be performed manually on Windows.
+First obtain the Windows 10 SDK, which contains the necessary signing utilities, from
+the Microsoft website.
+We need the `SignTool` utility which should have been installed somewhere like
+`C:\Program Files (x86)\Windows Kits\10\App Certification Kit`.
+Grab the Windows certificate files from CodeSigning on `juliasecure` and put them
+in the same directory as the executables.
+Open a Windows CMD window, `cd` to where all the files are, and run
+
+```
+set PATH=%PATH%;C:\Program Files (x86)\Windows Kits\10\App Certification Kit;
+signtool sign /f julia-windows-code-sign_2015.p12 /p "Juli@343" ^
```

Comment:
you shouldn't publicly post the password to a certificate

Reply:
... 😐 whoops

Reply:
to state the obvious, that password should now be changed on the image as well as here.

Reply:
Yes, both have been changed.